packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -1947,13 +1947,20 @@ class _OverlayPortalState extends State<OverlayPortal> {
     if (oldWidget.controller != widget.controller) {
       oldWidget.controller._attachTarget = null;
       _setupController(widget.controller);
+    } else if (widget.controller._attachTarget == null) {
+      _setupController(widget.controller);
     }
   }
 
   @override
-  void dispose() {
-    assert(widget.controller._attachTarget == this);
+  void deactivate() {
+    super.deactivate();
+    assert(widget.controller._attachTarget == null || widget.controller._attachTarget == this);
```

Comment:
I think the assert here should be `assert(widget.controller._attachTarget == this)`?

Reply:
Yes, that should be the case but when reattaching with global key, it was causing failures.
But i think since we are moving reattach in `activate` method, this can be removed.